components/optimizer/base.py

```diff
@@ -0,0 +1,40 @@
+import abc
+from common.minimodel import Group, Plan
+from components.selector import GroupInfo
+from typing import Dict
+
+Selection = Dict[Group, GroupInfo]
```

Comment:
LOL just saw this here, which contradicts my comment above, but what about L169 in the `run_scheduler.py` script below? I think maybe we should be looking up by `GroupID` and getting `Tuple[Group, GroupInfo]`? We may not be able to guarantee that a `Group` can be hashed.

Reply:
Fixed

Reply:
Fixed